src/Illuminate/Broadcasting/BroadcastEvent.php

```diff
@@ -63,16 +63,23 @@ public function fire(Job $job, array $data)
      */
     protected function getPayloadFromEvent($event)
     {
-        if (method_exists($event, 'broadcastWith')) {
-            return $event->broadcastWith();
-        }
-
         $payload = [];
 
-        foreach ((new ReflectionClass($event))->getProperties(ReflectionProperty::IS_PUBLIC) as $property) {
```

Comment:
Well because the trait adds properties on the event i though why not get them this why instead of checking if the class contains the trait and then get the socket. 

Because we are going to loop through the properties anyway.